assets/js/src/common/set_from_address_modal.tsx

```diff
@@ -41,6 +44,22 @@ const showSetFromAddressModal = async () => {
         />
       </div>
     ),
+    onInit: () => {
+      const saveButton = document.getElementById('mailpoet_set_from_address_modal_save') as HTMLInputElement;
+      const addressInput = document.getElementById('mailpoet_set_from_address_modal_address') as HTMLInputElement;
+      const addressValidator = jQuery(addressInput).parsley();
+
+      saveButton.addEventListener('click', async () => {
```

Comment:
This looks a bit odd to me. Again, I am not rejecting here, just asking for clarification. 
- why is this function async? it doesn't return anything, does it?
- you are combining react, jquery and native JS listeners in a single file, that doesn't seem right to me. There is technically nothing wrong with that. It just doesn't seem correct to me. Care to explain the reasoning?

Reply:
For the `async` here - the function uses `await` (below). I may have added the `await` in a later commit due to some rebases which is probably confusing, but in the end it needs `async`.

To explain React, jQuery, & JS (plus also old vs. new notices):
1. This is a modal which is a non-React component.
2. To get a body rendered by React in there I need to render the to string and pass it to the modal. This is done my `ReactDOMServer`. It's not new in our codebase, same logic is used in [confirm alert](https://github.com/mailpoet/mailpoet/blob/a8b31d1a745688acfff41ee8b23976803c02990f/assets/js/src/common/confirm_alert.jsx) or [NPS poll](https://github.com/mailpoet/mailpoet/blob/a8b31d1a745688acfff41ee8b23976803c02990f/assets/js/src/nps_poll.jsx) modals.
3. Since the result is a string, there are no hooks, no events, no "dynamic" React functionality. That's why any behavior must be done using native JS `addEventListener`.
4. I need form validation here and we use `Parsley`. Parsley is a library that extends jQuery which is why I need it to be able to call the `.parsley()` at all.
5. I also need notices but since I'm not in the React tree I don't have a way to use React context with the new notices. (Now, about this last point I'm not 100% sure, maybe there's a way how to interact with React notices from the modal but I guess it's not really important - because what we really need here is a rewrite of Modal component to React.)

Reply:
Do I understand it correctly that the reason for this combination of technologies is the modal? Why didn't you use this one then? Is there something wrong with it? https://github.com/mailpoet/mailpoet/blob/master/assets/js/src/common/modal/modal.jsx